server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -244,15 +251,28 @@ private boolean hasBonusPool(List<Pool> pools) {
     /*
      * Returns null if invalid
      */
-    private String getVirtQuantity(String virtLimit, long quantity) {
+    private String getVirtQuantity(String virtLimit, long masterPoolQuantity, boolean hostLimited) {
+        if (hostLimited && ("unlimited".equals(virtLimit) || masterPoolQuantity == UNLIMITED_QUANTITY)) {
+            return String.valueOf(UNLIMITED_QUANTITY);
+        }
+
         if ("unlimited".equals(virtLimit)) {
             return virtLimit;
         }
```

Comment:
We know that we should treat both Unmapped guest pools and Legacy virt bonus pools the same, so don't need the host_limited value at all in this method. We could also merge the if statement above with this one to be:
```
if ("unlimited".equals(virtLimit) || masterPoolQuantity == UNLIMITED_QUANTITY) {
    return "unlimited";
}
```
since all we care about is to return 'unlimited' if either virt_limit or the master pool is unlimited.

Reply:
Fixed.